docs/architecture/adr-014-namespace-id-size.md

```diff
@@ -0,0 +1,57 @@
+# ADR 14: Namespace ID Size
+
+## Changelog
+
+- 2023/2/17: initial draft
+
+## Context
+
+Namespace ID is currently an 8 byte slice. 8 bytes provides a maximum of 2^64 possible namespace IDs. In practice some namespace IDs are reserved for protocol use so the number of namespace IDs available to users is 2^56 - 2. Modifying the size of a namespace ID post-launch is a breaking change and has implications for the NMT, share encoding, etc. so we'd like to carefully consider the size of the namespace ID pre-launch.
```

Comment:
Do we want to consider increasing the number of reserved namespace IDs to 65536 or larger, if we also increase namespace ID size?

Reply:
Good question. Currently the reserved namespace range occupies:

```go
MinReservedNamespace := namespace.ID{0, 0, 0, 0, 0, 0, 0, 0}
MaxReservedNamespace = namespace.ID{0, 0, 0, 0, 0, 0, 0, 255}
```

and it currently uses 3 of the 256 range for transactions, PFB transactions, and max reserved namespace for reserved namespace padding. Given such low utilization (3/256 is 1% utilization) and without any proposals that increase utilization, I think it's safe to preserve the 256 range and expand at a later date if needed. Thoughts?

For completeness, the protocol also reserves these two:

```
TailPaddingNamespaceID = namespace.ID{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFE}
ParitySharesNamespaceID = namespace.ID{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
```

Reply:
I think this reserved namespace range is only for namespace version 0 right (assuming this adr is merged)? So if we add extra namespace versions we have to explicitly define the reserved namespace range for those

Reply:
Exactly. I expect the strategy you proposed to work if we wanted to increase the reserved namespace range via a namespace id version bump.

Reply:
With notable exceptions for tail padding and parity shares namespaces: https://github.com/celestiaorg/celestia-app/pull/1382#discussion_r1112247526